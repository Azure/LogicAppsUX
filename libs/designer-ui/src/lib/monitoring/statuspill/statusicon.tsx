import {
  getTheme,
  ISpinnerStyles,
  ITheme,
  registerOnThemeChangeCallback,
  removeOnThemeChangeCallback,
  Spinner,
  SpinnerSize,
} from '@fluentui/react';
import { useEffect, useState } from 'react';
import Constants from '../../constants';
import { Aborted } from './images/aborted';
import { Cancelled } from './images/cancelled';
import { Failed } from './images/failed';
import { Skipped } from './images/skipped';
import { Succeeded } from './images/succeeded';
import { SucceededWithRetries } from './images/succeededwithretries';
import { TimedOut } from './images/timedout';
import { Waiting } from './images/waiting';

export interface StatusIconProps {
  hasRetries: boolean;
  status: string;
}

const spinnerStyles: ISpinnerStyles = {
  root: {
    width: 24,
  },
};

export const StatusIcon: React.FC<StatusIconProps> = ({ hasRetries, status }) => {
  const [isInverted, setIsInverted] = useState(() => getTheme().isInverted);

  useEffect(() => {
    registerOnThemeChangeCallback(handleThemeChange);
    return () => {
      removeOnThemeChangeCallback(handleThemeChange);
    };
  }, []);

  const handleThemeChange = (theme: ITheme) => {
    setIsInverted(theme.isInverted);
  };

  if (status === Constants.STATUS.RUNNING) {
    return <Spinner size={SpinnerSize.medium} styles={spinnerStyles} />;
  }

  switch (status) {
    case Constants.STATUS.ABORTED:
      return <Aborted isInverted={isInverted} />;

    case Constants.STATUS.CANCELLED:
      return <Cancelled isInverted={isInverted} />;

    case Constants.STATUS.FAILED:
      return <Failed isInverted={isInverted} />;

    case Constants.STATUS.SKIPPED:
      return <Skipped isInverted={isInverted} />;

    case Constants.STATUS.SUCCEEDED:
      return hasRetries ? <SucceededWithRetries isInverted={isInverted} /> : <Succeeded isInverted={isInverted} />;

    case Constants.STATUS.TIMEDOUT:
      return <TimedOut isInverted={isInverted} />;

    case Constants.STATUS.WAITING:
      return <Waiting isInverted={isInverted} />;

    // Use 1x1 transparent image for unexpected status.
    default:
      return <img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" alt="" role="presentation" />;
  }
};
