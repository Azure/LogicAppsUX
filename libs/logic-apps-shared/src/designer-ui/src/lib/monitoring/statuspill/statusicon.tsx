import Constants from '../../constants';
import { Aborted } from './images/aborted';
import { Cancelled } from './images/cancelled';
import { Failed } from './images/failed';
import { Skipped } from './images/skipped';
import { Succeeded } from './images/succeeded';
import { SucceededWithRetries } from './images/succeededwithretries';
import { TimedOut } from './images/timedout';
import { Waiting } from './images/waiting';
import type { ISpinnerStyles } from '@fluentui/react';
import { Spinner, SpinnerSize } from '@fluentui/react';

type IconOpacity = `${number}%`;
export interface StatusIconProps {
  hasRetries: boolean;
  status: string;
  iconOpacity?: IconOpacity;
}

const spinnerStyles: ISpinnerStyles = {
  root: {
    width: 24,
  },
};

export const StatusIcon: React.FC<StatusIconProps> = ({ hasRetries, status, iconOpacity = '100%' }) => {
  if (status === Constants.STATUS.RUNNING) {
    return <Spinner size={SpinnerSize.medium} styles={spinnerStyles} />;
  }

  switch (status) {
    case Constants.STATUS.ABORTED:
      return <Aborted />;

    case Constants.STATUS.CANCELLED:
      return <Cancelled />;

    case Constants.STATUS.FAILED:
      return <Failed />;

    case Constants.STATUS.SKIPPED:
      return <Skipped />;

    case Constants.STATUS.SUCCEEDED:
      return <div style={{ opacity: iconOpacity }}>{hasRetries ? <SucceededWithRetries /> : <Succeeded />}</div>;

    case Constants.STATUS.TIMEDOUT:
      return <TimedOut />;

    case Constants.STATUS.WAITING:
      return <Waiting />;

    // Use 1x1 transparent image for unexpected status.
    default:
      return <img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" alt="" role="presentation" />;
  }
};
