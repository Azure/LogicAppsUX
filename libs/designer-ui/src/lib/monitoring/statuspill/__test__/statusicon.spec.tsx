import renderer from 'react-test-renderer';
import { StatusIcon } from '../statusicon';
import { describe, vi, beforeEach, afterEach, beforeAll, afterAll, it, test, expect } from 'vitest';
describe('lib/monitoring/statuspill/statusicon', () => {
  for (const { hasRetries, status } of [
    { hasRetries: false, status: 'Aborted' },
    { hasRetries: false, status: 'Cancelled' },
    { hasRetries: false, status: 'Failed' },
    { hasRetries: false, status: 'Running' },
    { hasRetries: false, status: 'Skipped' },
    { hasRetries: false, status: 'Succeeded' },
    { hasRetries: true, status: 'Succeeded' },
    { hasRetries: false, status: 'TimedOut' },
    { hasRetries: false, status: 'Unknown' },
    { hasRetries: false, status: 'Waiting' },
  ]) {
    it(`renders (status = ${status}, has retries = ${hasRetries})`, () => {
      const tree = renderer.create(<StatusIcon hasRetries={hasRetries} status={status} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  }
});
