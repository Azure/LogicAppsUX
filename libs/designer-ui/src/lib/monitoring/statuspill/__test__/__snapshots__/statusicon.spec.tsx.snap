// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lib/monitoring/statuspill/statusicon > renders (status = Aborted, has retries = false) 1`] = `
<svg
  fill="none"
  height="28"
  viewBox="0 0 28 28"
  width="28"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="14"
    cy="14"
    fill="#A4262C"
    r="13"
    stroke="#fff"
    strokeWidth="2"
  />
  <path
    d="M14.6221 14L19.3389 18.7236L18.7236 19.3389L14 14.6221L9.27637 19.3389L8.66113 18.7236L13.3779 14L8.66113 9.27637L9.27637 8.66113L14 13.3779L18.7236 8.66113L19.3389 9.27637L14.6221 14Z"
    fill="#fff"
  />
  <path
    d="M14.6221 14L14.4452 13.8234L14.2688 14L14.4452 14.1766L14.6221 14ZM19.3389 18.7236L19.5156 18.9004L19.6923 18.7238L19.5158 18.547L19.3389 18.7236ZM18.7236 19.3389L18.547 19.5158L18.7238 19.6923L18.9004 19.5156L18.7236 19.3389ZM14 14.6221L14.1766 14.4452L14 14.2688L13.8234 14.4452L14 14.6221ZM9.27637 19.3389L9.09959 19.5156L9.27624 19.6923L9.45302 19.5158L9.27637 19.3389ZM8.66113 18.7236L8.48423 18.547L8.30771 18.7238L8.48436 18.9004L8.66113 18.7236ZM13.3779 14L13.5548 14.1766L13.7312 14L13.5548 13.8234L13.3779 14ZM8.66113 9.27637L8.48436 9.09959L8.30771 9.27624L8.48423 9.45302L8.66113 9.27637ZM9.27637 8.66113L9.45302 8.48423L9.27624 8.30771L9.09959 8.48436L9.27637 8.66113ZM14 13.3779L13.8234 13.5548L14 13.7312L14.1766 13.5548L14 13.3779ZM18.7236 8.66113L18.9004 8.48436L18.7238 8.30771L18.547 8.48423L18.7236 8.66113ZM19.3389 9.27637L19.5158 9.45302L19.6923 9.27624L19.5156 9.09959L19.3389 9.27637ZM14.4452 14.1766L19.162 18.9003L19.5158 18.547L14.799 13.8234L14.4452 14.1766ZM19.1621 18.5469L18.5469 19.1621L18.9004 19.5156L19.5156 18.9004L19.1621 18.5469ZM18.9003 19.162L14.1766 14.4452L13.8234 14.799L18.547 19.5158L18.9003 19.162ZM13.8234 14.4452L9.09972 19.162L9.45302 19.5158L14.1766 14.799L13.8234 14.4452ZM9.45314 19.1621L8.83791 18.5469L8.48436 18.9004L9.09959 19.5156L9.45314 19.1621ZM8.83804 18.9003L13.5548 14.1766L13.201 13.8234L8.48423 18.547L8.83804 18.9003ZM13.5548 13.8234L8.83804 9.09972L8.48423 9.45302L13.201 14.1766L13.5548 13.8234ZM8.83791 9.45314L9.45314 8.83791L9.09959 8.48436L8.48436 9.09959L8.83791 9.45314ZM9.09972 8.83804L13.8234 13.5548L14.1766 13.201L9.45302 8.48423L9.09972 8.83804ZM14.1766 13.5548L18.9003 8.83804L18.547 8.48423L13.8234 13.201L14.1766 13.5548ZM18.5469 8.83791L19.1621 9.45314L19.5156 9.09959L18.9004 8.48436L18.5469 8.83791ZM19.162 9.09972L14.4452 13.8234L14.799 14.1766L19.5158 9.45302L19.162 9.09972Z"
    fill="#fff"
  />
</svg>
`;

exports[`lib/monitoring/statuspill/statusicon > renders (status = Cancelled, has retries = false) 1`] = `
<svg
  fill="none"
  height="24"
  viewBox="2 2 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="14"
    cy="14"
    fill="#605E5C"
    r="13"
    stroke="#fff"
    strokeWidth="2"
  />
  <path
    d="M14.6221 14L19.3389 18.7236L18.7236 19.3389L14 14.6221L9.27637 19.3389L8.66113 18.7236L13.3779 14L8.66113 9.27637L9.27637 8.66113L14 13.3779L18.7236 8.66113L19.3389 9.27637L14.6221 14Z"
    fill="#fff"
  />
  <path
    d="M14.6221 14L14.4452 13.8234L14.2688 14L14.4452 14.1766L14.6221 14ZM19.3389 18.7236L19.5156 18.9004L19.6923 18.7238L19.5158 18.547L19.3389 18.7236ZM18.7236 19.3389L18.547 19.5158L18.7238 19.6923L18.9004 19.5156L18.7236 19.3389ZM14 14.6221L14.1766 14.4452L14 14.2688L13.8234 14.4452L14 14.6221ZM9.27637 19.3389L9.09959 19.5156L9.27624 19.6923L9.45302 19.5158L9.27637 19.3389ZM8.66113 18.7236L8.48423 18.547L8.30771 18.7238L8.48436 18.9004L8.66113 18.7236ZM13.3779 14L13.5548 14.1766L13.7312 14L13.5548 13.8234L13.3779 14ZM8.66113 9.27637L8.48436 9.09959L8.30771 9.27624L8.48423 9.45302L8.66113 9.27637ZM9.27637 8.66113L9.45302 8.48423L9.27624 8.30771L9.09959 8.48436L9.27637 8.66113ZM14 13.3779L13.8234 13.5548L14 13.7312L14.1766 13.5548L14 13.3779ZM18.7236 8.66113L18.9004 8.48436L18.7238 8.30771L18.547 8.48423L18.7236 8.66113ZM19.3389 9.27637L19.5158 9.45302L19.6923 9.27624L19.5156 9.09959L19.3389 9.27637ZM14.4452 14.1766L19.162 18.9003L19.5158 18.547L14.799 13.8234L14.4452 14.1766ZM19.1621 18.5469L18.5469 19.1621L18.9004 19.5156L19.5156 18.9004L19.1621 18.5469ZM18.9003 19.162L14.1766 14.4452L13.8234 14.799L18.547 19.5158L18.9003 19.162ZM13.8234 14.4452L9.09972 19.162L9.45302 19.5158L14.1766 14.799L13.8234 14.4452ZM9.45314 19.1621L8.83791 18.5469L8.48436 18.9004L9.09959 19.5156L9.45314 19.1621ZM8.83804 18.9003L13.5548 14.1766L13.201 13.8234L8.48423 18.547L8.83804 18.9003ZM13.5548 13.8234L8.83804 9.09972L8.48423 9.45302L13.201 14.1766L13.5548 13.8234ZM8.83791 9.45314L9.45314 8.83791L9.09959 8.48436L8.48436 9.09959L8.83791 9.45314ZM9.09972 8.83804L13.8234 13.5548L14.1766 13.201L9.45302 8.48423L9.09972 8.83804ZM14.1766 13.5548L18.9003 8.83804L18.547 8.48423L13.8234 13.201L14.1766 13.5548ZM18.5469 8.83791L19.1621 9.45314L19.5156 9.09959L18.9004 8.48436L18.5469 8.83791ZM19.162 9.09972L14.4452 13.8234L14.799 14.1766L19.5158 9.45302L19.162 9.09972Z"
    fill="#fff"
  />
</svg>
`;

exports[`lib/monitoring/statuspill/statusicon > renders (status = Failed, has retries = false) 1`] = `
<svg
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="12"
    cy="12"
    fill="#fff"
    r="12"
  />
  <path
    d="M12 0C13.1 0 14.1583 0.145833 15.175 0.4375C16.2 0.720833 17.1583 1.125 18.05 1.65C18.9417 2.16667 19.75 2.79167 20.475 3.525C21.2083 4.25 21.8333 5.05833 22.35 5.95C22.875 6.84167 23.2792 7.8 23.5625 8.825C23.8542 9.84167 24 10.9 24 12C24 13.1 23.8542 14.1625 23.5625 15.1875C23.2792 16.2042 22.875 17.1583 22.35 18.05C21.8333 18.9417 21.2083 19.7542 20.475 20.4875C19.75 21.2125 18.9417 21.8375 18.05 22.3625C17.1583 22.8792 16.2 23.2833 15.175 23.575C14.1583 23.8583 13.1 24 12 24C10.9 24 9.8375 23.8583 8.8125 23.575C7.79583 23.2833 6.84167 22.8792 5.95 22.3625C5.05833 21.8375 4.24583 21.2125 3.5125 20.4875C2.7875 19.7542 2.1625 18.9417 1.6375 18.05C1.12083 17.1583 0.716667 16.2042 0.425 15.1875C0.141667 14.1625 0 13.1 0 12C0 10.9 0.141667 9.84167 0.425 8.825C0.716667 7.8 1.12083 6.84167 1.6375 5.95C2.1625 5.05833 2.7875 4.25 3.5125 3.525C4.24583 2.79167 5.05833 2.16667 5.95 1.65C6.84167 1.125 7.79583 0.720833 8.8125 0.4375C9.8375 0.145833 10.9 0 12 0ZM12.8 16H11.2V17.6H12.8V16ZM12.8 6.4H11.2V14.4H12.8V6.4Z"
    fill="#A4262C"
  />
</svg>
`;

exports[`lib/monitoring/statuspill/statusicon > renders (status = Running, has retries = false) 1`] = `
<div
  className="ms-Spinner root-110"
>
  <div
    className="ms-Spinner-circle ms-Spinner--medium circle-111"
  />
</div>
`;

exports[`lib/monitoring/statuspill/statusicon > renders (status = Skipped, has retries = false) 1`] = `
<svg
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="12"
    cy="12"
    fill="#fff"
    r="12"
  />
  <path
    d="M12 0C13.1 0 14.1625 0.141667 15.1875 0.425C16.2125 0.708333 17.1667 1.1125 18.05 1.6375C18.9333 2.1625 19.7417 2.7875 20.475 3.5125C21.2083 4.2375 21.8333 5.04583 22.35 5.9375C22.8667 6.82917 23.2708 7.7875 23.5625 8.8125C23.8542 9.8375 24 10.9 24 12C24 13.1083 23.8583 14.1708 23.575 15.1875C23.2917 16.2042 22.8875 17.1583 22.3625 18.05C21.8375 18.9417 21.2125 19.75 20.4875 20.475C19.7625 21.2 18.9542 21.825 18.0625 22.35C17.1708 22.875 16.2125 23.2792 15.1875 23.5625C14.1625 23.8458 13.1 23.9917 12 24C10.8917 24 9.82917 23.8583 8.8125 23.575C7.79583 23.2917 6.84167 22.8875 5.95 22.3625C5.05833 21.8375 4.25 21.2125 3.525 20.4875C2.8 19.7625 2.175 18.9542 1.65 18.0625C1.125 17.1708 0.720833 16.2125 0.4375 15.1875C0.154167 14.1625 0.00833333 13.1 0 12C0 10.9 0.141667 9.8375 0.425 8.8125C0.708333 7.7875 1.1125 6.83333 1.6375 5.95C2.1625 5.06667 2.7875 4.25833 3.5125 3.525C4.2375 2.79167 5.04583 2.16667 5.9375 1.65C6.82917 1.13333 7.7875 0.729167 8.8125 0.4375C9.8375 0.145833 10.9 0 12 0ZM19.2 12.8V11.2H4.8V12.8H19.2Z"
    fill="#605E5C"
  />
</svg>
`;

exports[`lib/monitoring/statuspill/statusicon > renders (status = Succeeded, has retries = false) 1`] = `
<div
  style={
    {
      "opacity": "100%",
    }
  }
>
  <svg
    fill="none"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      fill="#fff"
      r="12"
    />
    <path
      d="M12 0C13.1016 0 14.1641 0.144531 15.1875 0.433594C16.2109 0.714844 17.1641 1.11719 18.0469 1.64062C18.9375 2.16406 19.7461 2.79297 20.4727 3.52734C21.207 4.25391 21.8359 5.0625 22.3594 5.95312C22.8828 6.83594 23.2852 7.78906 23.5664 8.8125C23.8555 9.83594 24 10.8984 24 12C24 13.1016 23.8555 14.1641 23.5664 15.1875C23.2852 16.2109 22.8828 17.168 22.3594 18.0586C21.8359 18.9414 21.207 19.75 20.4727 20.4844C19.7461 21.2109 18.9375 21.8359 18.0469 22.3594C17.1641 22.8828 16.2109 23.2891 15.1875 23.5781C14.1641 23.8594 13.1016 24 12 24C10.8984 24 9.83594 23.8594 8.8125 23.5781C7.78906 23.2891 6.83203 22.8828 5.94141 22.3594C5.05859 21.8359 4.25 21.2109 3.51562 20.4844C2.78906 19.75 2.16406 18.9414 1.64062 18.0586C1.11719 17.168 0.710938 16.2109 0.421875 15.1875C0.140625 14.1641 0 13.1016 0 12C0 10.8984 0.140625 9.83594 0.421875 8.8125C0.710938 7.78906 1.11719 6.83594 1.64062 5.95312C2.16406 5.0625 2.78906 4.25391 3.51562 3.52734C4.25 2.79297 5.05859 2.16406 5.94141 1.64062C6.83203 1.11719 7.78906 0.714844 8.8125 0.433594C9.83594 0.144531 10.8984 0 12 0ZM19.0664 8.02734L17.4727 6.43359L9.75 14.1562L6.52734 10.9336L4.93359 12.5273L9.75 17.3438L19.0664 8.02734Z"
      fill="#428000"
    />
  </svg>
</div>
`;

exports[`lib/monitoring/statuspill/statusicon > renders (status = Succeeded, has retries = true) 1`] = `
<div
  style={
    {
      "opacity": "100%",
    }
  }
>
  <svg
    fill="none"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      fill="#fff"
      r="12"
    />
    <path
      d="M12 0C13.1016 0 14.1641 0.144531 15.1875 0.433594C16.2109 0.714844 17.1641 1.11719 18.0469 1.64062C18.9375 2.16406 19.7461 2.79297 20.4727 3.52734C21.207 4.25391 21.8359 5.0625 22.3594 5.95312C22.8828 6.83594 23.2852 7.78906 23.5664 8.8125C23.8555 9.83594 24 10.8984 24 12C24 13.1016 23.8555 14.1641 23.5664 15.1875C23.2852 16.2109 22.8828 17.168 22.3594 18.0586C21.8359 18.9414 21.207 19.75 20.4727 20.4844C19.7461 21.2109 18.9375 21.8359 18.0469 22.3594C17.1641 22.8828 16.2109 23.2891 15.1875 23.5781C14.1641 23.8594 13.1016 24 12 24C10.8984 24 9.83594 23.8594 8.8125 23.5781C7.78906 23.2891 6.83203 22.8828 5.94141 22.3594C5.05859 21.8359 4.25 21.2109 3.51562 20.4844C2.78906 19.75 2.16406 18.9414 1.64062 18.0586C1.11719 17.168 0.710938 16.2109 0.421875 15.1875C0.140625 14.1641 0 13.1016 0 12C0 10.8984 0.140625 9.83594 0.421875 8.8125C0.710938 7.78906 1.11719 6.83594 1.64062 5.95312C2.16406 5.0625 2.78906 4.25391 3.51562 3.52734C4.25 2.79297 5.05859 2.16406 5.94141 1.64062C6.83203 1.11719 7.78906 0.714844 8.8125 0.433594C9.83594 0.144531 10.8984 0 12 0ZM19.0664 8.02734L17.4727 6.43359L9.75 14.1562L6.52734 10.9336L4.93359 12.5273L9.75 17.3438L19.0664 8.02734Z"
      fill="#DB7500"
    />
  </svg>
</div>
`;

exports[`lib/monitoring/statuspill/statusicon > renders (status = TimedOut, has retries = false) 1`] = `
<svg
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="12"
    cy="12"
    fill="white"
    r="12"
  />
  <path
    d="M12 0C13.1016 0 14.1641 0.144531 15.1875 0.433594C16.2109 0.714844 17.1641 1.11719 18.0469 1.64062C18.9375 2.16406 19.7461 2.79297 20.4727 3.52734C21.207 4.25391 21.8359 5.0625 22.3594 5.95312C22.8828 6.83594 23.2852 7.78906 23.5664 8.8125C23.8555 9.83594 24 10.8984 24 12C24 13.1016 23.8555 14.1641 23.5664 15.1875C23.2852 16.2109 22.8828 17.168 22.3594 18.0586C21.8359 18.9414 21.207 19.75 20.4727 20.4844C19.7461 21.2109 18.9375 21.8359 18.0469 22.3594C17.1641 22.8828 16.2109 23.2891 15.1875 23.5781C14.1641 23.8594 13.1016 24 12 24C10.8984 24 9.83594 23.8594 8.8125 23.5781C7.78906 23.2891 6.83203 22.8828 5.94141 22.3594C5.05859 21.8359 4.25 21.2109 3.51562 20.4844C2.78906 19.75 2.16406 18.9414 1.64062 18.0586C1.11719 17.168 0.710938 16.2109 0.421875 15.1875C0.140625 14.1641 0 13.1016 0 12C0 10.8984 0.140625 9.83594 0.421875 8.8125C0.710938 7.78906 1.11719 6.83594 1.64062 5.95312C2.16406 5.0625 2.78906 4.25391 3.51562 3.52734C4.25 2.79297 5.05859 2.16406 5.94141 1.64062C6.83203 1.11719 7.78906 0.714844 8.8125 0.433594C9.83594 0.144531 10.8984 0 12 0ZM18 12H12V4.5H10.5V13.5H18V12Z"
    fill="#DB7500"
  />
</svg>
`;

exports[`lib/monitoring/statuspill/statusicon > renders (status = Unknown, has retries = false) 1`] = `
<img
  alt=""
  role="presentation"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
/>
`;

exports[`lib/monitoring/statuspill/statusicon > renders (status = Waiting, has retries = false) 1`] = `
<div
  aria-label="Run information loading"
  className="fui-Spinner r1k3z50n"
  role="progressbar"
>
  <span
    className="fui-Spinner__spinner rvgcg50 ___ruqntd0_k4fqbj0 fjamq6b f64fuq3 f1v3ph3m"
  >
    <span
      className="fui-Spinner__spinnerTail r1lregi5"
    />
  </span>
</div>
`;
